test/integration/graphql_test.lua

```diff
@@ -779,3 +779,134 @@ function g.test_middleware()
+            end
+            return args.arg.field
+        end
+
+        local graphql = require('cartridge.graphql')
+        local types = require('cartridge.graphql.types')
+
+        graphql.add_callback({
+            name = 'test_default_value',
+            args = {
+                arg = types.string,
+            },
+            kind = types.string,
+            callback = 'test.test_default_value',
+        })
+
+        graphql.add_callback({
+            name = 'test_default_list',
+            args = {
+                arg = types.list(types.string),
+            },
+            kind = types.string,
+            callback = 'test.test_default_list',
+        })
+
+        local input_object = types.inputObject({
+            name = 'default_input_object',
+            fields = {
+                field = types.string,
+            }
+        })
+
+        graphql.add_callback({
+            name = 'test_default_object',
+            args = {
+                arg = input_object,
+            },
+            kind = types.string,
+            callback = 'test.test_default_object',
+        })
+    ]])
+
+    t.assert_equals(
+        server:graphql({
+            query = [[
+                query($arg: String = "default_value") {
+                    test_default_value(arg: $arg)
+                }
+            ]],
+        variables = {}}
+        ).data.test_default_value, 'default_value'
+    )
+
+    t.assert_equals(
+        server:graphql({
+            query = [[
+                query($arg: String = "default_value") {
+                    test_default_value(arg: $arg)
+                }
+            ]],
+        variables = {arg = box.NULL}}
+        ).data.test_default_value, 'nil'
+    )
+
+    t.assert_equals(
+        server:graphql({
+            query = [[
+                query($arg: [String] = ["default_value"]) {
+                    test_default_list(arg: $arg)
+                }
+            ]],
+        variables = {}}
+        ).data.test_default_list, 'default_value'
+    )
+
+    t.assert_equals(
+        server:graphql({
+            query = [[
+                query($arg: [String] = ["default_value"]) {
+                    test_default_list(arg: $arg)
+                }
+            ]],
+        variables = {arg = box.NULL}}
+        ).data.test_default_list, 'nil'
+    )
+
+    t.assert_equals(
+        server:graphql({
+            query = [[
+                query($arg: default_input_object = {field: "default_value"}) {
+                    test_default_object(arg: $arg)
+                }
+            ]],
+        variables = {}}
+        ).data.test_default_object, 'default_value'
+    )
+
+    t.assert_equals(
+        server:graphql({
+            query = [[
+                query($arg: default_input_object = {field: "default_value"}) {
+                    test_default_object(arg: $arg)
+                }
+            ]],
+        variables = {arg = box.NULL}}
+        ).data.test_default_object, 'nil'
```

Comment:
Are you sure? I suppose it should return nothing (`nil`), not string `"nil"`

Reply:
Cartridge doesn't allow to return "nil" as a value. I return "nil" as string in my functions in such case - https://github.com/tarantool/cartridge/pull/920/files#diff-cf7848a656b50a8b4ce2cc926bef5205R790

Reply:
There is a root of such workaroundhttps://github.com/tarantool/cartridge/blob/1f4a8e026fefa2d8434bca4a2a2cde29750d59a9/cartridge/graphql.lua#L304